README.md

```diff
@@ -1343,18 +1344,145 @@ We run task **3** the output would be:
+
+> **Note:** The `$variable` syntax is currently not supported!
+
+> **Note:** Extended interpolation like `${variable:-default}` are currently unsupported!
+
+```toml
+RUST_BACKTRACE = 1
+BOOL_VALUE = true
+COMPOSITE = "${BOOL_VALUE} ${RUST_BACKTRACE}"
+```
+
+##### List
+
+`cargo-make` also supports lists, which are joined using `;` at runtime.
+
+```toml
+LIST_VALUE = [ "VALUE1", "VALUE2", "VALUE3" ]
+```
+
+##### Script
+
+`cargo-make` supports the use of simple scripts. The output of the said script will then determine the value of the environmental variable.
+
+The script's object has two additional arguments: `multiline` and `depends_on`. If `multiple` is set to `true`, the supplied script will be evaluated as a script with multiple lines. `depends_on` is a list of environmental variables this script depends on, which is taken into account during reordering if unset `cargo-make` will try to guess the variables used during reordering.
+
+> **Note:** This uses `sh` for script evaluation ([spec](https://pubs.opengroup.org/onlinepubs/9699919799/)), other runners like `duckscript`, `rust`, etc. are **not** supported.
+
+```toml
+EVALUATED_VAR = { script = ["echo SOME VALUE"] }
+```
+
+##### Decode Map
+
+`cargo-make` supports the use of mappings where a `source` is matched against a dictionary of possible `mapping`s, where each key of the `mapping` is compared against the evaluated `source` value. Should the key and `source` be the same, the corresponding value to the key will be the value of the environmental variable. If no key is matched, the `default_value` is used if provided. Otherwise, it will default to an empty string instead.
+
+```toml
+LIBRARY_EXTENSION = { source = "${CARGO_MAKE_RUST_TARGET_OS}", default_value = "unknown", mapping = {"linux" = "so", "macos" = "dylib", "windows" = "dll", "openbsd" = "so" } }
+```
+
+##### Path
+
+`cargo-make` supports the use of glob syntax to find all files and directories in a given directory. The list of files will be joined using `;` during execution.
+
+```toml
+PATH_GLOB = { glob = "./src/**/mod.rs", include_files = true, include_dirs = false, ignore_type = "git" }
+```
+
+##### Conditional
+
+`cargo-make` supports conditional variables, which are set to the `value` specified if the `condition` evaluates to true. To learn more about conditions, refer to [this chapter]
+
+##### Unset
+
+Variables can be unset, which will unset their value and reset it to an empty string.
+
+```toml
+VARIABLE = {unset = true}
+```
+
+#### Note about Ordering
+
+Environmental variables in `cargo-make`, create implicit dependencies between each other and are handled more in a declarative approach. Therefore ordering of variables is not necessarily the same between definition and evaluation.
```

Comment:
this entire section is too long and detailed. lets just explain that order is important but also dependencies may impact it.
i don't think we need to expand too much what happens if there is only basic implementation if its not really like that now.
this includes the native/simple sub sections.

however, i would add that this logic is only for env block and not for task env block.

Reply:
My bad, this only works on env blocks. I thought that `merge_env` was getting called in `environment::set_env_for_config`, but I was mistaken.

My question would be: do we want re-ordering at the last stage? So when we run an actual task (this would mean we would need to call `merge_env` in `run_task_with_options`, if I understood the code correctly.

Reply:
na, i think we can document that we are only doing it for the env makefile level and not task level if it simplifies things.

Reply:
@sagiegurari I have a specific question about this. Let's say I have 3 tasks:

Makefile:
sets `env1` to `1`

Task A:
sets `env1` to `2`

Task B:
sets `env1` to `3`

Task C:
get `env1`
depends_on: [Task A, Task B]

Would `env1` for Task C be `3` in that case? It looked like env variables are not rolled back once a previous task sets them.

Reply:
yes, it would be 3. they are never rolled back. 
the whole point is that tasks allow you to modify the env and affect the next tasks that come.
for example one task may setup the env based on a condition block.

Reply:
I gotcha that makes sense.